### PR TITLE
Add empty entry on nullable fields that will render as select inputs

### PR DESCRIPTION
**Pull request by michha**
It was painfull to manually add the empty option to the baked views. So I added code that would check the schema and if the field is nullable, the empty option is added.

### CODE REVIEW

**Comment by dereuromark** [comment]
Sounds reasonable to me.

File: src/Template/Bake/Element/form.ctp [hunk 1]
```diff
@@ -58,9 +58,17 @@ $fields = collection($fields)
                 continue;
             }
             if (isset($keyFields[$field])) {
+                $fieldData = $schema->column($field);
+                if (array_key_exists('null', $fieldData) && $fieldData['null'] == 1) {
```
Comment (markstory) [inline src/Template/Bake/Element/form.ctp]: You can use !empty() instead here.

File: src/Template/Bake/Element/form.ctp [hunk 2]
```diff
@@ -58,9 +58,17 @@ $fields = collection($fields)
...
+%>
+            echo $this->Form->input('<%= $field %>', ['options' => $<%= $keyFields[$field] %>, 'empty' => true]);
+<%
+                }
+                else {
```
Comment (markstory) [inline src/Template/Bake/Element/form.ctp]: Should be `} else {`.

**Comment by michha** [comment]
Fixed both issues

**Comment by lorenzo** [comment]
Good job, thanks!

**Comment by dereuromark** [comment]
Actually, even for not NULL fields it is bad to pre-select. Instead the text should be slightly different.
I have always been using this for the last years: https://github.com/dereuromark/cakephp-setup/blob/master/Console/Templates/setup/views/form.ctp#L52-L55

**Comment by jhli** [comment]
I do agree. Two ways to go:
- Don't pre-select and blank for NULL and NOT NULL
- Don't pre-select and different messages for NULL and NOT NULL. For NOT NULL it should say something like "Please select..." and for not null something like "You may select..." 

I am fine with both but pre-selecting NULL foreign keys is bad from a utility point of view and pre-selecting NULL and NOT NULL values is bad from an ergonomics point of view (users get confused if they already chose something or it it was pre-selected)

**Pull request merged**